lib/webrick/httpservlet/filehandler.rb

```diff
@@ -324,7 +324,7 @@ def get_handler(req, res)
       end
 
       def set_filename(req, res)
-        res.filename = @root.dup
+        res.filename = @root.b
```

Comment:
```
--- filehandler.rb.old	2020-04-09 09:49:57.536780938 +0800
+++ filehandler.rb	2020-04-09 09:58:41.387264503 +0800
@@ -520,6 +520,7 @@
           s =  "<TR><TD class=\"name\"><A HREF=\"#{HTTPUtils::escape(name)}#{query if name.end_with?('/')}\">#{HTMLUtils::escape(dname)}</A></TD>"
           s << "<TD class=\"mtime\">" << (time ? time.strftime("%Y/%m/%d %H:%M") : "") << "</TD>"
           s << "<TD class=\"size\">" << (size >= 0 ? size.to_s : "-") << "</TD></TR>\n"
+          s.force_encoding 'ASCII-8BIT'
           res.body << s
         }
         res.body << "</TBODY></TABLE>"
```

Reply:
I tried with CJK filename in CJK directory (as DocumentRoot), but `s.force_encoding 'ASCII-8BIT'` is not required.
How to reproduce the case to need `s.force_encoding 'ASCII-8BIT'`?

Reply:
require 'webrick'

d="\u3042"
Dir.mkdir(d) if not File.exist? d
port = 8080
config = { :DocumentRoot => "." , :Port => port }
s=WEBrick::HTTPServer.new(config)
Thread.new {s.start }
url = "http://127.0.0.1:#{port}/%E3%81%82/" # from firefox
`curl #{url} `
sleep 20

Reply:
```ruby
require 'webrick'

d="\u3042"
Dir.mkdir(d) if not File.exist? d
`echo CJK_file > #{d}/#{d}.txt`
port = 8080
config = { :DocumentRoot => "." , :Port => port }
s=WEBrick::HTTPServer.new(config)
Thread.new {s.start }
p url = "http://127.0.0.1:#{port}/%E3%81%82/%E3%81%82.txt" # OK                                
`curl #{url} `
p url = "http://127.0.0.1:#{port}/%E3%81%82/" # ERR
`curl #{url}`
sleep 30
```